src/utils/daos.c

```diff
@@ -1289,13 +1289,13 @@ main(int argc, char *argv[])
 	/* argv[1] is RESOURCE or "help" or "version";
 	 * argv[2] if provided is a resource-specific command
 	 */
-	if (argc < 2 || strcmp(argv[1], "help") == 0) {
+	if (argc >= 2 && strcmp(argv[1], "version") == 0) {
```

Comment:
Minor suggestion: invoke help_hdlr and return nonzero status from the process when user provides command lines such as
daos help blah blah blah .... (endless until shell tells you that the line is too long)
daos version blah blah blah ... (endless)

while still protecting against the segfault for partial commands such as "daos container" or "daos pool" or "daos obj" with no further arguments.

something like this might work
if (argc == 2 && strcmp(argv[1], "version") == 0)
else if (argc < 2 || (argc == 2 && strcmp(argv[1], "help") == 0))
else if argc <= 2 {
  dargs.ostream = stdout;
  help_hdlr(argc, argv, &dargs);
  return 2;
} else if ...

Reply:
@bfaccini that sounds right to me. Could simplify it to: `else if (argc < 2 || strcmp(argv[1], "help") == 0)`

Reply:
@kjacque , right !! Why make it simple ;-)